src/main/java/com/cvsgo/repository/ProductCustomRepositoryImpl.java

```diff
@@ -58,72 +69,142 @@ public List<SearchProductQueryDto> searchByFilter(User user,
             .fetch();
     }
 
-    public Optional<ProductDetailResponseDto> findByProductId(User user, Long productId) {
+    public Optional<SearchProductDetailQueryDto> findByProductId(User loginUser, Long productId) {
         return Optional.ofNullable(queryFactory
-            .select(new QProductDetailResponseDto(
-                product.as("product"),
-                product.manufacturer.as("manufacturer"),
-                productLike.as("productLike"),
-                productBookmark.as("productBookmark")
+            .select(new QSearchProductDetailQueryDto(
+                product.id,
+                product.name,
+                product.price,
+                product.imageUrl,
+                manufacturer.name,
+                productLike,
+                productBookmark
             ))
             .from(product)
             .leftJoin(productLike)
-            .on(productLike.product.id.eq(productId).and(productLike.user.eq(user)))
+            .on(productLike.product.eq(product).and(productLikeUserEq(loginUser)))
             .leftJoin(productBookmark)
-            .on(productBookmark.product.id.eq(productId).and(productBookmark.user.eq(user)))
+            .on(productBookmark.product.eq(product).and(productBookmarkUserEq(loginUser)))
+            .leftJoin(manufacturer).on(product.manufacturer.eq(manufacturer))
             .where(product.id.eq(productId))
             .fetchFirst());
     }
 
-    private BooleanExpression eqProductLikeUser(User user) {
+    public List<ConvenienceStoreEventQueryDto> findConvenienceStoreEventsByProductId(
+        Long productId) {
+        return queryFactory.select(new QConvenienceStoreEventQueryDto(
+                sellAt.convenienceStore.id,
+                sellAt.convenienceStore.name,
+                event))
+            .from(sellAt)
+            .leftJoin(event).on(sellAt.product.eq(event.product)
+                .and(sellAt.convenienceStore.eq(event.convenienceStore)))
+            .leftJoin(convenienceStore).on(sellAt.convenienceStore.eq(convenienceStore))
+            .where(sellAt.product.id.eq(productId))
+            .fetch();
+    }
+
+    private static List<OrderSpecifier> sortBy(ProductSortBy sortBy, NumberPath<Double> score,
+        NumberPath<Double> avgRating) {
+        List<OrderSpecifier> orderSpecifiers = new ArrayList<>();
+        if (sortBy != null) {
+            switch (sortBy) {
+                case SCORE -> {
+                    orderSpecifiers.add(score.desc());
+                    orderSpecifiers.add(avgRating.desc());
+                    orderSpecifiers.add(product.likeCount.desc());
+                    orderSpecifiers.add(product.createdAt.desc());
+                }
+                case RATING -> {
+                    orderSpecifiers.add(avgRating.desc());
+                    orderSpecifiers.add(product.likeCount.desc());
+                    orderSpecifiers.add(product.createdAt.desc());
```

Comment:
리뷰 평점이 동일하다면 리뷰 개수가 더 많은 것이 먼저 오도록 이렇게 하는 건 어떤가요?ㅎㅎ
```suggestion
                case RATING -> {
                    orderSpecifiers.add(avgRating.desc());
                    orderSpecifiers.add(reviewCount.desc()); // 추가
                    orderSpecifiers.add(product.likeCount.desc());
                    orderSpecifiers.add(product.createdAt.desc());
```

Reply:
좋아요~ default로는 `순위 -> 평균 평점 -> 리뷰 개수 -> 상품 좋아요수 -> 최신순` 으로 정렬하겠습니다😉